drivers/hmis/app/models/hmis/access_group.rb

```diff
@@ -12,39 +12,40 @@ class Hmis::AccessGroup < ApplicationRecord
   acts_as_paranoid
   has_paper_trail
 
-  has_many :access_group_members, class_name: '::Hmis::AccessGroupMember'
-  has_many :users, through: :access_group_members
+  has_many :access_controls, class_name: '::Hmis::AccessControl', inverse_of: :access_group
+  has_many :user_access_controls, through: :access_controls
+  has_many :users, through: :user_access_controls
+  has_many :roles, through: :access_controls
+  has_many :access_groups, through: :access_controls
 
-  has_many :group_viewable_entities, class_name: 'GrdaWarehouse::GroupViewableEntity'
-  has_many :data_sources, through: :group_viewable_entities, source: :entity, source_type: 'GrdaWarehouse::DataSource'
-  has_many :organizations, through: :group_viewable_entities, source: :entity, source_type: 'GrdaWarehouse::Hud::Organization'
-  has_many :projects, through: :group_viewable_entities, source: :entity, source_type: 'GrdaWarehouse::Hud::Project'
-  has_many :project_access_groups, through: :group_viewable_entities, source: :entity, source_type: 'GrdaWarehouse::ProjectAccessGroup'
+  has_many :group_viewable_entities, class_name: 'Hmis::GroupViewableEntity'
+  has_many :organizations, through: :group_viewable_entities, source: :entity, source_type: 'Hmis::Hud::Organization'
+  has_many :projects, through: :group_viewable_entities, source: :entity, source_type: 'Hmis::Hud::Project'
 
   validates_presence_of :name
 
   scope :general, -> do
     all
   end
 
-  # If you can edit, you can also view
   scope :viewable, -> do
-    where(scope: ['view', 'edit'])
+    joins(:roles).merge(Hmis::Role.with_viewable_permissions)
   end
 
   scope :editable, -> do
-    where(scope: ['edit'])
+    joins(:roles).merge(Hmis::Role.with_editable_permissions)
   end
```

Comment:
We may want the distinction between viewable and editable at this level, but it seems unnecessary to me.  Is this so we can more easily determine if the page should be read-only or is this just left over from the previous mechanism?

Reply:
I think it's just left over, I don't see why we would need it. Each mutation should eventually be checking for specific permission(s).

Reply:
Agreed that this is not a necessary distinction and is better abandoned. IIRC I've left it in here to preserve the functionality of the `editable` and `viewable` methods on `Hmis::User`, and perhaps in other places. I didn't feel like I had a deep enough understanding of the intent of those methods to make decisions on their functionality without input from you all, so I erred on the side of preserving functionality that was already there.

Overall I think dropping the editable/viewable distinction in all cases in favor of checking for the presence of specific permissions is best, but I'm open to being mistaken about that. As with the above, I wanted to get input from you all before making that decision.

Reply:
Based on [conversation in slack](https://greenriver.slack.com/archives/C033YNSN0Q1/p1677531465874729) on 2/27/23, we're going to leave the `editable` and `viewable` distinction until we migrate completely over to ACLs